beater/mysqlbeat.go

```diff
@@ -189,40 +201,43 @@ LoopQueries:
 		}
 
 		// Populate the two-columns event
-		if bt.queryTypes[index] == queryTypeTwoColumns {
+		if query["type"] == queryTypeTwoColumns {
 			twoColumnEvent = common.MapStr{
 				"@timestamp": common.Time(dtNow),
 				"type":       queryTypeTwoColumns,
+				"server":     server,
 			}
 		}
 
 	LoopRows:
 		for rows.Next() {
 
-			switch bt.queryTypes[index] {
+			switch query["type"] {
 			case queryTypeSingleRow, queryTypeSlaveDelay:
 				// Generate an event from the current row
-				event, err := bt.generateEventFromRow(rows, columns, bt.queryTypes[index], dtNow)
+				event, err := bt.generateEventFromRow(rows, columns, query["type"], dtNow)
 
 				if err != nil {
 					logp.Err("Query #%v error generating event from rows: %v", index+1, err)
 				} else if event != nil {
+					event["server"] = server
```

Comment:
as mentioned elsewhere, "server" is probably better as "hostname" in the event (applies below as well)